daisy_workflows/linux_common/bootstrap.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/sh
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -xeu
+
+GetMetadataAttribute() {
+    attribute="$1"
+
+    url="$metadata_url/v1/instance/attributes/$attribute"
+    attribute_value=$(curl -H "Metadata-Flavor: Google" -X GET "$url")
+}
+
+DebianInstallGoogleApiPythonClient() {
+    logger -p daemon.info "Status: Installing google-api-python-client"
+
+    apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -q -y install python3-pip
+    pip3 install -U google-api-python-client google-cloud-storage
+}
+
+DebianInstallNetaddrPythonLibrary() {
+    logger -p daemon.info "Status: Installing netaddr python module"
+
+    apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -q -y install python3-netaddr
+}
+
+GetAccessToken() {
+    url="$metadata_url/v1/instance/service-accounts/default/token"
+    response=$(curl -H "Metadata-Flavor: Google" -X GET "$url")
+
+    access_token_value=$(echo "$response" | cut -d '"' -f 4)
+    token_type_value=$(echo "$response" | grep -o '"token_type":".*"' | cut -d '"' -f 4)
+
+    TOKEN="$token_type_value $access_token_value"
+}
+
+GetBucketContent() {
+    bucket_name="$1"
+    bucket_path="$2"
+    token="$3"
+    storage_url="https://www.googleapis.com/storage"
+
+    url="$storage_url/v1/b/$bucket_name/o?prefix=$bucket_path"
+
+    logger -p daemon.info "Status: Bucket listing with $bucket_path prefix: $url"
+    response=$(curl -H "Metadata-Flavor: Google" -H "Authorization: $token" -X GET "$url")
+
+    name_attributes=$(echo "$response" | tr "," "\\n" | grep name)
+    bucket_content=$(echo "$name_attributes" | tr " " "\\n" | grep "$bucket_path" | sed -e 's/\"//g')
+}
+
+SaveBucketFile() {
+    bucket_target="$1"
+    bucket_file="$2"
+    file_dest="$3"
+
+    url="https://storage.googleapis.com/$bucket_target/$bucket_file"
+
+    dir=$(dirname "$file_dest")
+    [ ! -d "$dir" ] && mkdir -p "$dir"
+
+    logger -p daemon.info "Status: Bucket save: $url => $file_dest"
+    curl -H "Metadata-Flavor: Google" -H "Authorization: $token" -X GET "$url" -o "$file_dest"
+}
+
+CheckPython3Installation() {
```

Comment:
Is there a case for this right now? It wouldn't work on EL6 as far as I know anyway. Maybe we should remove it as to not cause problems later on.

Reply:
Yes, this is needed for image configuration test. I just executed that daisy workflow using a RHEL 6 image and it works. RHEL 6 already has the EPEL repository installed and everything is fine.

Reply:
Oh, I didn't know RHEL 6 had py3 in EPEL. But cool.